bolt/test/AArch64/hook-fini.s

```diff
@@ -0,0 +1,94 @@
+## Test the different ways of hooking the fini function for instrumentation (via
+## DT_FINI and via DT_FINI_ARRAY). We test the latter for both PIE and non-PIE
+## binaries because of the different ways of handling relocations (static or
+## dynamic).
+## All tests perform the following steps:
+## - Compile and link for the case to be tested
+## - Some sanity-checks on the dynamic section and relocations in the binary to
+##   verify it has the shape we want for testing:
+##   - DT_FINI or DT_FINI_ARRAY in dynamic section
+##   - No relative relocations for non-PIE
+## - Instrument
+## - Verify generated binary
+# REQUIRES: system-linux,bolt-runtime
+
+# RUN: %clang  %s -Wl,-q -o %t.exe
```

Comment:
Please add %cflags. And lets add -fPIC -pie in pic tests explicitly. The same for the runtime test please

Reply:
I got this mostly working. There is an issue though with the no-pie test: without pie and when not linking against shared libraries, no dynamic section is generated. This means that the current implementation doesn't support these binaries. I think this is fine but it means we cannot write tests for the non-pie case. Do you see a solution for this?

Reply:
If adding shared library helps you can add one, like I've done in test/X86/unreachable.c .
There is file in test/common/stub.c which you can use to do
RUN: %clang %cflags %p/../Inputs/stub.c -fPIC -pie -shared -o %t.so
And link against it.

Reply:
Thanks for that @yota9! Nice trick :) Tests are updated.

Reply:
No problem :) Could you please also rebase your changes? I still can see my changes in the patch :)